### PR TITLE
Date time

### PR DESCRIPTION
**Pull request by filbert-c**
1.Date is inside "Time" class
2.Period is "Time"
3.Currently debugging GUI test cases[null exceptions everywhere], before pushing to refactor.

### CODE REVIEW

**Comment by filbert-c** [comment]
#22

**Comment by filbert-c** [comment]
updated fixed test cases

File: src/main/java/seedu/address/logic/parser/Parser.java [hunk 1]
```diff
@@ -27,9 +27,10 @@
             Pattern.compile("(?<keywords>\\S+(?:\\s+\\S+)*)"); // one or more keywords separated by whitespace
 
     private static final Pattern PERSON_DATA_ARGS_FORMAT = // '/' forward slashes are reserved for delimiter prefixes
-            Pattern.compile("(?<name>[^/]+)"
-            		+ "(( (?<isPhonePrivate>p?)t/(?<phone>[^/]+))?)"
-            		+ "(( (?<isEmailPrivate>p?)d/(?<email>[^/]+))?)"
+            Pattern.compile("(?<name>[^;]+)"
+            		+ "(( (?<isPhonePrivate>p?)t;(?<phone>[^;]+)))"
+            		+ "(( et;(?<period>[^;]+))?)"
+            		+ "(( (?<isEmailPrivate>p?)d;(?<email>[^/]+))?)"
```
Comment (YesYouKenSpace) [inline src/main/java/seedu/address/logic/parser/Parser.java:33]: Our regex changed a lot already in the main branch

File: src/main/java/seedu/address/logic/parser/Parser.java [hunk 2]
```diff
@@ -102,6 +103,7 @@ private Command prepareAdd(String args){
             return new AddCommand(
                     matcher.group("name"),
                     matcher.group("phone")==null?" ":matcher.group("phone"),
+                    matcher.group("period")==null?"23:59":matcher.group("period"),
```
Comment (YesYouKenSpace) [inline src/main/java/seedu/address/logic/parser/Parser.java:106]: I understand the constraint with natty. But can we still pass in an empty string, but have a layer of check for empty string before passing to natty? If empty string then don't let it be processed by natty.

File: src/main/java/seedu/address/model/task/Time.java [hunk 3]
```diff
@@ -1,42 +1,84 @@
...
+     *
+     * @author Alok Chaudhary, Filbert(with the help of Java regex converter)
+     */
+    public static final String DATE_VALIDATION_REGEX = "^(?:(?:31(\\/|-|\\.)(?:0?[13578]|1[02]|"
+            + "(?:Jan|Mar|May|Jul|Aug|Oct|Dec)))\\1|(?:(?:29|30)(\\/|-|\\.)(?:0?[1,3-9]|1[0-2]|"
+            + "(?:Jan|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec))\\2))(?:(?:1[6-9]|[2-9]\\d)?\\d{2})$|^"
+            + "(?:29(\\/|-|\\.)(?:0?2|(?:Feb))\\3(?:(?:(?:1[6-9]|[2-9]\\d)?(?:0[48]|[2468][048]|[13579][26])|"
+            + "(?:(?:16|[2468][048]|[3579][26])00))))$|^(?:0?[1-9]|1\\d|2[0-8])(\\/|-|\\.)(?:(?:0?[1-9]|"
+            + "(?:Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep))|(?:1[0-2]|(?:Oct|Nov|Dec)))"
+            + "\\4(?:(?:1[6-9]|[2-9]\\d)?\\d{2})$";
```
Comment (YesYouKenSpace) [inline src/main/java/seedu/address/model/task/Time.java:30]: Wow I wanted to refactor this, then I don't even know where to start... hahas forget it.

**Review by YesYouKenSpace — CHANGES_REQUESTED** [review]
I think you need to git pull first. I foresee you will be pekchek with the amount of conflicts hahahas. Just push your working branch to remote repo. Then maybe we can work on it together? There might be changes needed in View (@MCHowl) and Edit because of this.

**Pull request merged**